### PR TITLE
[cpuid] Update to 0.6.2

### PR DESCRIPTION
**Pull request by chausner**
Updates cpuid to 0.6.2

* [x]  Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
* [x]  SHA512s are updated for each updated download
* [x]  The "supports" clause reflects platforms that may be fixed by this new version
* [x]  Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
* [x]  Any patches that are no longer applied are deleted from the port's directory.
* [x]  The version database is fixed by rerunning ./vcpkg x-add-version --all and committing the result.
* [x]  Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/cpuid/portfile.cmake [hunk 1]
```diff
@@ -1,12 +1,8 @@
-if(VCPKG_TARGET_IS_WINDOWS)
-    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
-endif()
```
Comment (chausner) [inline ports/cpuid/portfile.cmake:3]: Removed this since https://github.com/anrieff/libcpuid/pull/156 added support for shared builds on Windows.

**Review by JonLiu1993 — APPROVED** [review]

File: ports/cpuid/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/cpuid/portfile.cmake:3]: ..at the price of `EXPORT_ALL_SYMBOLS`.

Comment (JonLiu1993) [reply ports/cpuid/portfile.cmake:3]: Yes, I also noticed that he used `EXPORT_ALL_SYMBOLS`, which is not allowed for us vcpkg, but he added it to the upstream code.
https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#do-not-add-cmake_windows_export_all_symbols

Comment (BillyONeal) [reply ports/cpuid/portfile.cmake:3]: I believe our stance on this is we won't add `EXPORT_ALL_SYMBOLS`, and strongly recommend against its use, but if that's what upstream does that's what upstream does.

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**